surface/include/pcl/surface/impl/convex_hull.hpp

```diff
@@ -214,16 +213,15 @@ pcl::ConvexHull<PointInT>::performReconstruction2D (PointCloud &hull, std::vecto
   }
 
   int num_vertices = qh num_vertices;
-  hull.points.resize (num_vertices);
-  memset (&hull.points[0], hull.size (), sizeof (PointInT));
+  hull.clear();
+  hull.resize(num_vertices);
 
   vertexT * vertex;
   int i = 0;
 
   std::vector<std::pair<int, Eigen::Vector4f>, Eigen::aligned_allocator<std::pair<int, Eigen::Vector4f> > > idx_points (num_vertices);
-  idx_points.resize (hull.size ());
-  memset (&idx_points[0], hull.size (), sizeof (std::pair<int, Eigen::Vector4f>));
-
+  const auto initPair = std::make_pair(hull.size(), Eigen::Vector4f());
```

Comment:
We can replace these 3 lines with just 1: using the default value to the constructor along with the size:

`LoongLongType idx_points(hull.size(), {hull.size(), Eigen::Vector4f::Zeros()});`